lib/hashie/extensions/deep_merge.rb

```diff
@@ -3,7 +3,7 @@ module Extensions
     module DeepMerge
       # Returns a new hash with +self+ and +other_hash+ merged recursively.
       def deep_merge(other_hash, &block)
-        copy = dup
+        copy = _deep_dup(self)
```

Comment:
I was wondering why we create a dup all the time? What is the reason behind this?

Reply:
The reason for this is due to the semantics between `#merge` and `#merge!`. The plain method does not mutate the receiver, the bang-method does.